lib/streamlit/delta_generator.py

```diff
@@ -690,26 +798,26 @@ def _arrow_add_rows(self, data=None, **kwargs):
         return self
 
 
-def _maybe_melt_data_for_add_rows(data, delta_type, last_index):
-    import pandas as pd
+DFT = TypeVar("DFT", bound="DataFrameCompatible")
 
-    # For some delta types we have to reshape the data structure
-    # otherwise the input data and the actual data used
-    # by vega_lite will be different and it will throw an error.
-    if (
-        delta_type in DELTA_TYPES_THAT_MELT_DATAFRAMES
-        or delta_type in ARROW_DELTA_TYPES_THAT_MELT_DATAFRAMES
-    ):
-        if not isinstance(data, pd.DataFrame):
-            data = type_util.convert_anything_to_df(data)
 
-        if type(data.index) is pd.RangeIndex:
-            old_step = _get_pandas_index_attr(data, "step")
+def _maybe_melt_data_for_add_rows(
+    data: "DataFrameCompatible",
```

Comment:
Isn't it possible to use the `DFT` TypeVar here as well?

Reply:
Good spot! That was clearly what I was planning on, but I must have forgot. Updated.